enhancements/worker-latency-profile/worker-latency-profile.md

```diff
@@ -0,0 +1,450 @@
+---
+title: worker-latency-profile
+authors:
+  - "@harche"
+reviewers:
+  - "@rphillips"
+  - "@sttts"
+  - "@soltysh"
+approvers:
+  - "@rphillips"
+  - "@sttts"
+  - "@soltysh"
+creation-date: 2021-12-07
+last-updated: 2021-12-07
+status: implementable
+see-also:
+  - https://github.com/kubernetes-sigs/kubespray/blob/master/docs/kubernetes-reliability.md
+  - https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#controllermanagerconfig
+replaces:
+superseded-by:
+---
+
+# WorkerLatencyProfile
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+To make sure the Openshift cluster keeps running optimally where network latency between the control plane and the worker nodes may not always be at its best (e.g. Edge cases), we can tweak the frequency of the status updates done by the `Kubelet` and the corresponding reaction times of the `Kube Controller Manager` and `Kube API Server`.
+
+## Motivation
+
+To accommodate the higher network latency (such as Edge cases) cluster may experience, we need to adjust how frequently the `Kubelet` updates its status and how components such as `Kube Controller Manager` and `Kube API Server`react to those status updates.
+
+The main motivation of this enhancement is to allow setting relevant arguments for these critical components in a more controlled manner instead of letting the users directly modify them manually which will make the cluster unsupported. This will also eliminate any room for manual errors that could lead to unscheduled downtime.
+
+### Goals
+
+* Enable Openshift cluster to fine tune the reliability in medium to high latency scenarios by specifying simple `WorkerLatencyProfile` at day-0 as well as day-2.
+
+### Non-Goals
+
+* Modify the existing `Kubelet`, `Kube Controller Manager` or `Kube API Server` code in any way.
```

Comment:
```suggestion
* Modify the existing `Kubelet`, `Kube Controller Manager` or `Kube API Server` configuration in any way.
```

Reply:
I actually meant `code` only. IMHO, we are indeed changing the configuration of the `Kubelet`, `Kube Controller Manager` or `Kube API Server` by changing the values of the respective arguments (e.g. `--node-monitor-grace-period` arg of `Kube Controller Manager`)